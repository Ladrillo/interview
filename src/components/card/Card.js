import React, { Component } from 'react';
import StyledButton from '../../styled/button';
import './Card.css';


export default function Card({ id, name, image, birthday, homeworld, onEdit }) {
    return (
        <div className='card'>
            <div className='card-content'>
                <div className='card-name'>{name}</div>
                <img src={`http://localhost:3008/${image}`} alt='profile' />
                <p>
                    <span>Birthday:</span>
                    <span>{birthday}</span>
                </p>
                <p>
                    <span>Homeworld:</span>
                    <span>{homeworld}</span>
                </p>
            </div>
            <div className='card-buttons'>
                <StyledButton onClick={() => onEdit(id)}>edit</StyledButton>
            </div>
        </div>
    );
}
