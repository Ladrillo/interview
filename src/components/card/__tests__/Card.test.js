import Card from '../Card';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';


describe('Card', () => {
    const id          = 'id';
    const name        = 'Luke Skywalker';
    const birthday    = 'AAA000X';
    const homeworld   = 'Tatooine';
    const isFav       = false;
    const onEdit      = () => console.log('editing...');
    const onToggleFav = () => console.log('toggling fav...');
Â 
    it('renders correctly', () => {
        const rendered = renderer.create(
            <Card
                id          = {id}
                name        = {name}
                birthday    = {birthday}
                homeworld   = {homeworld}
                isFav       = {isFav}
                onEdit      = {onEdit}
                onToggleFav = {onToggleFav}
            />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});