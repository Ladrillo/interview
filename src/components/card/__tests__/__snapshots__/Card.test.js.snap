exports[`Card renders correctly 1`] = `
<div
  className="card">
  <div
    className="card-content">
    <div
      className="card-name">
      Luke Skywalker
       
      <span
        className="sc-bwzfXH fYeCzG"
        onClick={[Function]}>
        â™¡
      </span>
    </div>
    <img
      alt="profile"
      src="http://localhost:3008/undefined" />
    <p>
      <span>
        Birthday:
      </span>
      <span>
        AAA000X
      </span>
    </p>
    <p>
      <span>
        Homeworld:
      </span>
      <span>
        Tatooine
      </span>
    </p>
  </div>
  <div
    className="card-buttons">
    <div
      className="sc-bdVaJa Tteup"
      onClick={[Function]}>
      edit
    </div>
  </div>
</div>
`;
